Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -303,6 +319,19 @@ public struct PackageContainerConstraint<T: PackageContainerIdentifier>: CustomS
                 return false
             }
         }
+
+        public var hashValue: Int {
+            switch self {
+            case .versionSet(let set):
+                return 0x4 ^ set.hashValue
+
+            case .revision(let str):
+                return 0x2 ^ str.hashValue
+
+            case .unversioned:
+                return 0x1
+            }
```

Comment:
Same thing here about random constants. Hopefully this all goes away at some point and Swift synthesizes good hash values.